frontend/js/pages/DetailBattle.js

```diff
@@ -6,88 +6,99 @@ import styled from 'styled-components';
   width: 30px;
   padding-left: 5px;
 `;
 
 class DetailBattle extends Component {
   componentDidMount() {
-    const { setBattle, match } = this.props;
+    const { setBattle, match, battles } = this.props;
     const { id } = match.params;
-    setBattle(id);
+    if (battles === undefined) {
+      setBattle(id);
+    }
   }
 
   render() {
-    const { battle, isLoading, user } = this.props;
-    const { id, creator_team, opponent_team, winner } = battle;
-    if (isLoading) return <>Loading</>;
+    const { battle, session } = this.props;
+    if (!battle) return <>Loading</>;
+
+    const { id, creator, opponent, winner } = battle;
     return (
       <>
         <h1>Battle nº {id}</h1>
         <div>
           <p>
-            <StyledTitle>Players: </StyledTitle>
-            {creator_team.trainer} <span>VS</span> {opponent_team.trainer}
+            <Title>Players: </Title>
+            <TrainersInBattle creator={creator.trainer} opponent={opponent.trainer} />
           </p>
 
           <p>
-            <StyledTitle>Winner: </StyledTitle>
+            <Title>Winner: </Title>
             {winner || '?'}
-            {winner === user.email && <StyledIcon alt="winner" src={party} />}
+            {winner === session.email && <Icon alt="winner" src={party} />}
           </p>
-          <StyledContainer>
-            {creator_team.team.map((pokemon, index) => (
-              <div key={pokemon.id}>
-                <StyledTitle>Round {index + 1}</StyledTitle>
-                <StyledRoundContainer>
-                  <PokemonCard pokemon={pokemon} trainer={creator_team.trainer} />
-                  <StyledVersus>VS</StyledVersus>
+          <Container>
+            {creator.team.map((pokemon, index) => (
+              <div key={pokemon}>
+                <Title>Round {index + 1}</Title>
+                <RoundContainer>
+                  <PokemonCard pokemon={pokemon} trainer={creator.trainer} />
+                  <VSTag>VS</VSTag>
                   <PokemonCard
-                    pokemon={winner ? opponent_team.team[index] : undefined}
-                    trainer={opponent_team.trainer}
+                    pokemon={opponent.team ? opponent.team[index] : undefined}
+                    trainer={opponent.trainer}
                   />
-                </StyledRoundContainer>
+                </RoundContainer>
               </div>
             ))}
-          </StyledContainer>
+          </Container>
         </div>
       </>
     );
   }
 }
 
 DetailBattle.propTypes = {
+  battles: PropTypes.object,
   battle: PropTypes.object,
   setBattle: PropTypes.func,
-  isLoading: PropTypes.bool,
   match: PropTypes.object,
-  user: PropTypes.object,
+  session: PropTypes.object,
 };
 
-const mapStateToProps = (state) => ({
-  battle: state.battles.battle,
-  isLoading: state.battles.isLoading,
-  user: state.user,
-});
+const mapStateToProps = (state, { match }) => {
+  const { id } = match.params;
+  if (state.battles.battles) {
```

Comment:
It's better to have a consistent return on mapStateToProps. Always returns the same keys.